### DISCUSSION TITLE
Towards a Python3-only version 1.0

### DISCUSSION THREAD

**Discussion by erezsh** [lark-parser/lark#908] [General]
We've had a few discussions on the topic before, but now I've grown convinced that Lark should drop Python 2 support for the main development branch.

We can still backport support for critical fixes, but Lark is pretty stable already.

Since this is a major backwards-incompatible change, it might be a good opportunity to also move into a backwards-incompatible version 1, that will also fix a lot of the interface inconsistencies that Lark has accumulated over the years.

Is anyone interested in helping with this?

Any other thoughts on the subject?

**Comment by ThatXliner** [comment]
Do you plan to use pyupgrade?

**Reply by erezsh** [reply]
I don't know. I might give it a try.

**Comment by chanicpanic** [comment]
Sounds like a good approach. I am interested in helping out with the transition.

**Comment by erezsh** [comment]
@chanicpanic Your offer to help is much appreciated.

I've created an initial branch+PR: https://github.com/lark-parser/lark/pull/925

I will probably make more changes in the coming week. 

The remaining tasks, that I'm aware of:

Tasks remaining:

- Merge .pyi files into the code, and remove them.
- Update the interface to be more self-consistent. I already have some thoughts in mind, and I'm open to ideas.
- Improve the code with Python3 features, where possible (optional/low priority)
- Can you think of anything else?

**Reply by chanicpanic** [reply]
Great, I'm willing to start working on merging the type stubs. I can't think of anything else at the moment, but I will let you know if I come up with any ideas as I go through the code.

**Reply by ThatXliner** [reply]
You want to make a [project board](https://github.com/lark-parser/lark/projects) for this?

**Reply by erezsh** [reply]
I'm not sure it's necessary, but let's give it a go: https://github.com/lark-parser/lark/projects/1
